readers-frontend/src/pages/[topLevelCategory]/[...path].astro

```diff
@@ -324,3 +332,35 @@ if (category.categories_name === "muse" && page.currentPage === 0) {
     />
   </div>
 </Layout>
+
+<script define:vars={{ scrollCategories }}>
+  const desktopNav = document.querySelector("#desktopNav");
+  const mobileHeader = document.querySelector("#mobileHeader");
+  window.onscroll = function () {
```

Comment:
`addEventListener` instead of `onscroll`. Advantages: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

It would probably help with moving the code to a separate component too.